src/main/java/edu/rpi/legup/ui/proofeditorui/treeview/TreeView.java

```diff
@@ -1,756 +1,881 @@
+    }
+
+    /**
+     * Called when a tree puzzleElement is added to the tree
+     *
+     * @param treeElement TreeElement that was added to the tree
+     */
+    @Override
+    public void onTreeElementAdded(TreeElement treeElement) {
+        if (treeElement.getType() == NODE) {
+            addTreeNode((TreeNode) treeElement);
+        }
+        else {
+            addTreeTransition((TreeTransition) treeElement);
+        }
+        repaint();
+    }
+
+    /**
+     * Called when a tree puzzleElement is removed from the tree
+     *
+     * @param element TreeElement that was removed to the tree
+     */
+    @Override
+    public void onTreeElementRemoved(TreeElement element) {
+        if (element.getType() == NODE) {
+            TreeNode node = (TreeNode) element;
+            TreeNodeView nodeView = (TreeNodeView) viewMap.get(node);
+
+            nodeView.getParentView().setChildView(null);
+            removeTreeNode(node);
+        }
+        else {
+            TreeTransition trans = (TreeTransition) element;
+            TreeTransitionView transView = (TreeTransitionView) viewMap.get(trans);
+
+            //unlock ancestor elements if case rule deleted
+            Rule rule = trans.getRule();
+            for(TreeNode node : trans.getParents()){
+                //only if the last case of a case rule will be deleted
+                if (rule instanceof CaseRule && node.getChildren().isEmpty()) {
+                    //System.out.println("TRANSISTION UNLOCKING");
+                    CaseRule caseRule = (CaseRule)rule;
+                    //set dependent elements to be modifiable by ancestors (if not dependent on others)
+                    List<TreeNode> ancestors = node.getAncestors();
+                    //ancestors.add(node);
+                    for(TreeNode ancestor : ancestors) {
+                        if (!(ancestor.getParent() == null)) {
+                            for (PuzzleElement pelement : caseRule.dependentElements(node.getBoard(), trans.getSelection())) {
+                                PuzzleElement oldElement = ancestor.getParent().getBoard().getPuzzleElement(pelement);
+                                oldElement.setCasesDepended(oldElement.getCasesDepended() - 1);
+                                Point loc = ((GridCell)oldElement).getLocation();
+                                //System.out.println("("+loc.x+","+loc.y+") to "+oldElement.getCasesDepended());
+                                if (oldElement.getCasesDepended() == 0) {
+                                    //set modifiable if started modifiable
+                                    boolean modifiable = tree.getRootNode().getBoard().getPuzzleElement(oldElement).isModifiable();
+
+                                    //unmodifiable if already modified
+                                    TreeNode modNode = ancestor.getParent().getParents().get(0);
+                                    while(modNode.getParent()!=null){
```

Comment:
There should be spaces between values and operators.

Reply:
added